File: truss-compiler/src/main/java/truss/compiler/support/ArgumentOption.java
package truss.compiler.support;

public class ArgumentOption extends Argument {
    private String value;

    public ArgumentOption(String option, String description, boolean mandatory) {
        super(option, description, mandatory);
    }

    @Override
    public boolean allowMultiple() {
        return false;
    }

    @Override
    public boolean hasParameter() {
        return true;
    }

    public String getValue() {
        return value;
    }

    @Override
    public void setValue(String value) {
        this.value = value;
    }
}

File: Truss.Compiler/Support/ArgumentOption.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Support {
    public class ArgumentOption : Argument {
        public ArgumentOption(string option, string description, bool mandatory)
            : base(option, description, mandatory) {
        }

        public override bool AllowMultiple {
            get { return false; }
        }

        public override bool HasParameter {
            get { return true; }
        }

        public string Value { get; private set; }

        public override void SetValue(string value) {
            Value = value;
        }
    }
}
